provisioning/miniprov/hare_mp/main.py

```diff
@@ -38,6 +38,88 @@
 from hare_mp.store import ConfStoreProvider
 
 
+# returns 1 if v1 is bigger
+#         -1 if v2 is bigger
+#         0 if equal
+def version_compare(v1: str, v2: str) -> int:
+
+    # This will split both the versions by '.'
+    arr1 = v1.split(".")
+    arr2 = v2.split(".")
+    n = len(arr1)
+    m = len(arr2)
+
+    # converts to integer from string
+    arr1 = [int(i) for i in arr1]
```

Comment:
Versions can have letters, e.g. 2.0.32-rc

Here are the version names from Consul history: https://releases.hashicorp.com/consul/

Can we give a try to something like this instead of re-inventing the wheel: https://pypi.org/project/cmp_version/ ?

Reply:
@mssawant Can you remind me what is the result of this discussion? Are we good to add external library to check the versions? And do we need to check the versions given we already have that check at RPM layer?

@SwapnilGaonkar7 My rule of a thumb would be
- If we can't leverage third party library but need to keep your hand-made version check, then we need unit tests that will prove that your version comparison work.
- If we can leverage 3rd party library, let's our own implementation of the same algorithms.

Reply:
@SwapnilGaonkar7 any updates on this PR?

Reply:
@knekrasov It was decided that as consul is common 3rd party utility used by many components, all the checks related to it must be done in common location(mostly provisioner) and not in specific component like hare.
So I think we can safely close(discard) this PR.